### PR TITLE
Add indeterminate checkbox

### PR DESCRIPTION
**Pull request by kerstin97**
## Description

Add `CheckboxIndeterminate` icon to `MuiCheckbox`

## Example

-   [x] I have verified if my change requires an example

## Screenshots/screencasts

| Before   | After   |
| -------- | ------- |
| image     | image    |

## Changeset

-   [x] I have verified if my change requires a changeset

## Related tasks and documents
https://vivid-planet.atlassian.net/browse/COM-1285

### CODE REVIEW

File: project-words.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
 NestJS
 Embeddables
+indeterminated
```
Comment (jamesricky) [inline project-words.txt:3]: What is this? 😅

Comment (kerstin97) [reply project-words.txt]: This is for the spellcheck, the word is not in the dictionary, so we must add it in the project-words.txt: https://github.com/vivid-planet/comet/pull/2838

Comment (jamesricky) [reply project-words.txt]: What spellcheck? Is this really needed for anything? 😅

Comment (jamesricky) [reply project-words.txt]: Actually, now that I think about it, `indeterminated` isn't a word, the correct word would be `indeterminat`. 

Could you please update the icons name to fix that?
Then it matches MUIs prop name and should be valid in any spellcheck (why ever that is relevant 🤷🏼‍♂️)

Comment (kerstin97) [reply project-words.txt]: I've changed the name to `CheckboxIndeterminate`, also see docs from MUI https://v5.mui.com/material-ui/api/checkbox/#props

File: packages/admin/admin-icons/src/CheckboxIndeterminate.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { SvgIcon, SvgIconProps } from "@mui/material";
+
+export default function CheckboxIndeterminate(props: SvgIconProps) {
```
Comment (johnnyomair) [inline packages/admin/admin-icons/src/CheckboxIndeterminate.tsx:3]: Why can't we use a SVG for this, like we do for the other icons?

Comment (jamesricky) [reply packages/admin/admin-icons/src/CheckboxIndeterminate.tsx:3]: They aren't really Icons to be used as standard icons, the checkbox icons require custom svg-elements with classes, so we can style them correctly, depending on their state (like `disabled`) same with e.g., [CheckboxChecked](https://github.com/vivid-planet/comet/blob/main/packages/admin/admin-icons/src/CheckboxChecked.tsx). 

Arguably they shouldn't even be inside the `@comet/admin-icons` package, but directly in the `@comet/admin-theme` package where they are used, as there isn't really another use-case for these icons, as far as i am aware.

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**